packages/ra-core/src/i18n/TranslationProvider.tsx

```diff
@@ -13,7 +12,6 @@ import { I18nProvider } from '../types';
 interface Props {
     locale?: string;
     i18nProvider: I18nProvider;
-    children: ReactElement<any>;
```

Comment:
I don't understand that change

Reply:
`FunctionComponent` has `children` defined with as a `ReactNode`
```ts
interface FunctionComponent<P = {}> {
    (props: PropsWithChildren<P>, context?: any): ReactElement | null;
    ...
}

type PropsWithChildren<P> = P & { children?: ReactNode };
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/24f1d0c82da2d898acd03fbb3e692eba3c431f82/types/react/index.d.ts#L517-L518
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/24f1d0c82da2d898acd03fbb3e692eba3c431f82/types/react/index.d.ts#L773

By defining children in the `TranslationProvider` props we are restricting it further from what React accepts. This is incorrect. Context Provider's accept anything really as their children.

The change is also needed because TypeScript complains about `TranslationProvider` usage in `CoreAdminContext` because `children` there is the same loose type instead of the strict `ReactElement`.